preview/src/components/select/variants/main/mod.rs

```diff
@@ -3,17 +3,60 @@ use dioxus_primitives::select::{
     Select, SelectGroup, SelectGroupLabel, SelectItemIndicator, SelectList, SelectOption,
     SelectTrigger,
 };
+use strum::{EnumCount, IntoEnumIterator};
+
+#[derive(Debug, Clone, Copy, PartialEq, strum::EnumCount, strum::EnumIter, strum::Display)]
+enum Fruit {
+    Apple,
+    Banana,
+    Orange,
+    Strawberry,
+    Watermelon,
+}
+
+impl Fruit {
+    const fn emoji(&self) -> &'static str {
+        match self {
+            Fruit::Apple => "🍎",
+            Fruit::Banana => "🍌",
+            Fruit::Orange => "🍊",
+            Fruit::Strawberry => "🍓",
+            Fruit::Watermelon => "🍉",
+        }
+    }
+}
+
 #[component]
 pub fn Demo() -> Element {
+    let fruits = Fruit::iter().enumerate().map(|(i, f)| {
+        rsx! {
+            SelectOption::<Option<Fruit>> {
+                index: i,
+                class: "select-option",
+                value: Some(f),
+                display: f.to_string(),
+                {format!("{} {f}", f.emoji())}
+                SelectItemIndicator {
+                    svg {
+                        class: "select-check-icon",
+                        view_box: "0 0 24 24",
+                        xmlns: "http://www.w3.org/2000/svg",
+                        path { d: "M5 13l4 4L19 7" }
+                    }
+                }
+            }
+        }
+    });
+
     rsx! {
         document::Link {
             rel: "stylesheet",
             href: asset!("/src/components/select/variants/main/style.css"),
         }
-        Select {
+        Select::<Option<Fruit>> {
```

Comment:
@ealmloff, now each component has generic specialization, but they don't affect each other; so, if you specify the wrong types (e.g. `String` for `Select`, but `Option<Fruit>` for `SelectGroup`), you'll have pretty scary runtime errors. I can avoid it by creating some macro or generic builder, but maybe you have an opinion on this.